core/src/main/java/com/radixdlt/consensus/vertexstore/VertexStoreState.java

```diff
@@ -89,25 +89,18 @@
  * <p>In future, we'd like to move to having a separate vertex store, responsible for maintaining
  * its own state.
  */
-@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
 @Immutable
 public final class VertexStoreState {
   private static final Logger logger = LogManager.getLogger();
 
   private final VertexWithHash root;
   private final HighQC highQC;
-  // TODO: collapse the following two
-  private final ImmutableList<VertexWithHash> vertices;
-  private final ImmutableMap<HashCode, VertexWithHash> idToVertex;
+  private final ImmutableSet<VertexWithHash> vertices;
 
   private VertexStoreState(
-      HighQC highQC,
-      VertexWithHash root,
-      ImmutableMap<HashCode, VertexWithHash> idToVertex,
-      ImmutableList<VertexWithHash> vertices) {
+      HighQC highQC, VertexWithHash root, ImmutableSet<VertexWithHash> vertices) {
```

Comment:
It's not immediately clear to me without looking at the rest of the code if the `vertices` also contains the root? Perhaps we could tweak its name or add a comment?

EDIT: Looks like it's explicitly `nonRootVertices`, so perhaps we should call it that?